#BVall Analysis

Sequenced 3 * 24 individuals 


Pyrad: 
```
==** parameter inputs for pyRAD version 3.0.61  **======================== affected step ==
./                        ## 1. Working directory                                 (all)
              ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)
              ## 3. Loc. of barcode file (if not line 18)             (s1)
vsearch                   ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)
muscle                    ## 5. command (or path) to call muscle                  (s3,s7)
TGCAG                     ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)
20                         ## 7. N processors (parallel)                           (all)
2                         ## 8. Mindepth: min coverage for a cluster              (s4,s5)
4                         ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)
.93                       ## 10. Wclust: clustering threshold as a decimal        (s3,s6)
ddrad                       ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)
2                         ## 12. MinCov: min samples in a final locus             (s7)
50                         ## 13. MaxSH: max inds with shared hetero site          (s7)
BV4                 ## 14. Prefix name for final output (no spaces)         (s7)
==== optional params below this line ===================================  affected step ==
                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)
                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)
                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)
/gdc_home4/alexjvr/Bombina/BV4/demultiplexed/                       ## 18.opt.: loc. of de-multiplexed data                      (s2)
                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)
                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)
                       ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)
                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)
                       ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)
                       ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)
                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)
                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)
                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)
                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)
                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)
*                       ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)
                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)
50                       ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)
                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)
                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)
                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)
                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)
                       ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)
==== optional: list group/clade assignments below this line (see docs) ==================

```

Chose 93% clustering threshold after some optimisation runs. 

Lenient with Depth statistics. 

1. Description of the data



2. Summary statistics: Dataset1

  - average heterozygosity
  
  - heterozygosity per region
  
  - Het of old vs new ponds
  
  - comparison between regions
  
  - LD
  
  - HWE
  
  - Fis

3. Population structure 

3.1. IBD

3.2. look for barriers?

3.3. Fst
  - with different MAC
  
3.5. Distance tree

3.6. TESS

3.7. DAPC


#Dataset 1: little missing data


Missingness in unfiltered vcf file: 

```
vcftools --vcf BV234.vcf --missing-indv

mawk '!/IN/' out.imiss | cut -f5 > totalmissing

gnuplot << \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin( $1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
```

![alt_txt][Fig1.BVall]
[Fig1.BVall]:https://cloud.githubusercontent.com/assets/12142475/20074745/db091042-a531-11e6-8411-6c0fb0876a92.png



Step 1: remove loci with >0.5 missing data
```
vcftools --vcf BV234.vcf --max-missing 0.5 --recode --recode-INFO-all --out BV234.0.5.s1

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf BV234.vcf
        --recode-INFO-all
        --max-missing 0.5
        --out BV234.0.5.s1
        --recode

Eighth Header entry should be INFO: INFO    
After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 72766 out of a possible 1065852 Sites
Run Time = 37.00 seconds

```

![alt_txt][BV234.2]
[BV234.2]:https://cloud.githubusercontent.com/assets/12142475/20074828/24dd0944-a532-11e6-964a-8259a417ce3d.png


Step2: 

MAC 3

```
vcftools --vcf BV234.s1.names.vcf --mac 3 --recode --recode-INFO-all --out BV234.s2.mac3

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf BV234.s1.names.vcf
	--recode-INFO-all
	--mac 3
	--out BV234.s2.mac3
	--recode

After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 32066 out of a possible 72766 Sites
Run Time = 2.00 seconds
```


Step3: 

Thin 500

```
vcftools --vcf BV234.s2.mac3.recode.vcf --thin 500 --recode --recode-INFO-all --out BV234.s3.mac3.thin500

VCFtools - v0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf BV234.s2.mac3.recode.vcf
	--recode-INFO-all
	--thin 500
	--out BV234.s3.mac3.thin500
	--recode

After filtering, kept 72 out of 72 Individuals
Outputting VCF file...
After filtering, kept 17432 out of a possible 32066 Sites
Run Time = 1.00 seconds
```



Step4: 

Convert to plink & calculate OHet > 0.5

```
plink --file BV234.s3 --hardy
```

Load into R. Sort by O.Het. Write table with only Snps to remove

```
BV.het <- read.table("plink.hwe", header=T)

sort.BV.het <- BV.het[order(-BV.het$O.HET.),]

Ohet.BV.0.5 <- sort.BV.het[which(sort.BV.het$O.HET.>0.5),]

Ohet.Final <- Ohet.BV.0.5$SNP
write.table(Ohet.Final, "Ohet.Final", quote=F, row.names=F)
```

remove these 1268 loci using plink

```
plink --file BV234.s3 --exclude Ohet.Final --recodeA --recode --out BV234.s4
```

Calculate LD between loci 
```
plink --file BV234.s4 --r2



```

Convert with pgdspider
