#BV analyses for Report

March 2017

##Questions

1. Population structure

2. Barriers to gene-flow

3. Colonisation of artificial ponds


##Datasets

####1. Natural pops only


Remove all individuals from artificial ponds in vcftools 

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/SumStats
```
vcftools --vcf BV.71.1665.FINAL.vcf --remove names.Art.ponds --recode --recode-INFO-all --out BV46.Naturalponds.1665.vcf
```

46 indivs

7 pops

1665 loci

88.2% genotyping rate



####2. Full dataset

71 indivs

10 pops

1665 loci

87.9% genotyping rate


##Analyses

###1. Summary statistics

Dataset2

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/SumStats/AvgHet

####1. AvgHet

AvgHet was calculated before: https://github.com/alexjvr1/BombinaVariegata/blob/57751523f9e249e66d9a3e1933664958a109c771/BombinaFinalAnalyses.md

Remove IBA05_12A from the dataset. Calculate the avg Het and SD per population.
This is in the excel sheet: Bombina.Tables_1070303.xlsx

Plot data from all 71 populations. Colour Artificial and Natural ponds. 
```
BV71.AvgHet.forplot <- read.table("AvgHet.71.forplot", header=T)
colnames(BV71.AvgHet.forplot)
BV71.AvgHet.forplot$pop <- factor(BV71.AvgHet.forplot$pop, levels=BV71.AvgHet.forplot$pop)

q <- qplot(pop, AvgHet, fill=factor((PondType), levels=c("Artificial", "Natural")),
data=BV71.AvgHet.forplot, geom="boxplot")

pdf("Fig1.BV71.AvgHet.pdf")
q + theme_bw()+theme(legend.title=element_blank()) +theme(axis.text.x = element_text(angle = 90, hjust = 1))
dev.off()
```

![alt_txt][Fig1]
[Fig1]:https://cloud.githubusercontent.com/assets/12142475/23587537/bf9b6fb0-01ae-11e7-8418-4ba09fb1f658.png


####2. Fis & Heterozygosity

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/SumStats

calculate avg het and Fis overall. Per pop we can calculate H0 and He, but I couldn't get hierfstat to work on the individual pop files
```
library(adegenet)
library(hierfstat)
popnames.BV71 <- pop.BV.71$V2   ##list of popnames that can be read by seppop
popnames.BV71 <- gsub("1.", "", popnames.BV71)
popnames.BV71 <- gsub("2.", "", popnames.BV71)
popnames.BV71 <- gsub("3.", "", popnames.BV71)
popnames.BV71 <- gsub("4.", "", popnames.BV71)
popnames.BV71

BV.71.new <- BV.71  ##new genind file
popnames.BV71.factor <- as.factor(popnames.BV71)  
BV.71.new@pop <- popnames.BV71.factor  #rename pops
BV.71.new

stats.BRU <- summary(BV10pops.adegenet$BRU)   ##calculate summary stats
stats.ZIN <- summary(BV10pops.adegenet$ZIN)
stats.ZIV <- summary(BV10pops.adegenet$ZIV)
stats.NAG <- summary(BV10pops.adegenet$NAG)
stats.WIL <- summary(BV10pops.adegenet$WIL)
stats.HOP <- summary(BV10pops.adegenet$HOP)
stats.CHR <- summary(BV10pops.adegenet$CHR)
stats.IBA <- summary(BV10pops.adegenet$IBA)
stats.HOC <- summary(BV10pops.adegenet$HOC)
stats.UNT <- summary(BV10pops.adegenet$UNT)

mean(stats.BRU$Hobs)
mean(stats.ZIN$Hobs)
mean(stats.ZIV$Hobs)
mean(stats.NAG$Hobs)
mean(stats.WIL$Hobs)
mean(stats.HOP$Hobs)
mean(stats.CHR$Hobs)
mean(stats.IBA$Hobs)
mean(stats.HOC$Hobs)
mean(stats.UNT$Hobs)

hier.BRU <- genind2hierfstat(BV10pops.adegenet$BRU)   ##change to hierfstat format
hier.ZIN <- genind2hierfstat(BV10pops.adegenet$ZIN)
hier.ZIV <- genind2hierfstat(BV10pops.adegenet$ZIV)
hier.NAG <- genind2hierfstat(BV10pops.adegenet$NAG)
hier.WIL <- genind2hierfstat(BV10pops.adegenet$WIL)
hier.HOP <- genind2hierfstat(BV10pops.adegenet$HOP)
hier.CHR <- genind2hierfstat(BV10pops.adegenet$CHR)
hier.IBA <- genind2hierfstat(BV10pops.adegenet$IBA)
hier.HOC <- genind2hierfstat(BV10pops.adegenet$HOC)
hier.UNT <- genind2hierfstat(BV10pops.adegenet$UNT)

hier.BRU.stats <- basic.stats(hier.BRU, diploid=T, digits=2)  ##keep getting an error with these commands
```

All info in table in excel sheet. 


####3. Fst

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/SumStats

```
BV.71 <- read.structure("BV.71.1665.str")
BV.71

pop.BV.71 <- read.table("pops4pgdspider", header=F)  ##make sure the populations are numbered "01.DE.B", etc.
BV.10pops.factor <- as.factor(pop.BV.71$V2)
BV.71@pop <- BV.10pops.factor

hier.BV <- genind2hierfstat(BV.71)

BV.fst <- pairwise.fst(BV.71, pop=NULL, res.type=c("dist", "matrix"))

m <- as.matrix(BV.fst)
m2 <- melt(m)[melt(upper.tri(m))$value,]
names(m2)<- c("c1","c2", "distance")


##remove 0.00 at the diagonals from the plot to get the correct scale

BV.fst[BV.fst == 0] <- NA

library(gplots)

shadesOfGrey <- colorRampPalette(c("grey100", "grey0"))  ##define the colourpalette. 

Dend <- read.table("heatmap.popcolours", header=T)  ##list of colour names for each population based on R colour palatte. In alphabetical order (as in genind file)
Dend.Colours <- as.character(Dend$colours.pop)

BV.popnames <- c("BRU", "ZIN", "ZIV", "NAG", "WIL", "HOP", "CHR", "IBA", "HOC", "UNT")

pdf("Fig2.BV71.Fst.pdf")
par(oma=c(1,1,2,1))
heatmap.2(as.matrix(BV.fst), na.rm=T, trace="none", RowSideColors=Dend.Colours, ColSideColors=Dend.Colours, col=shadesOfGrey, labRow=BV.popnames, labCol=F, key.ylab=NA, key.xlab=NA, key.title="Fst Colour Key", keysize=0.9, main="Pairwise Fst BV71: 10pops, 3regions, 1665loci")  ##RowSideColors is for the dendrogram on the row, ColSideColors for the upper dendrogram. Colour order should be the same as the input. The pop order is alphabetical in the output. 
par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")

popnames.all <- as.character(c("BRU", "ZIN", "Middle", "East"))
legend("bottom", popnames.all, xpd = TRUE, horiz = TRUE, inset = c(0, 0), bty="o", pch=15, col=c("darkorange3", "darkorange2", "darkorchid1", "deepskyblue1"), title="Regions")
dev.off()

```
Figure 2

![alt_txt][Fig2]
[Fig2]:https://cloud.githubusercontent.com/assets/12142475/23588193/24b8afce-01ba-11e7-9dcc-0d806ad6abae.png





###2. Population structure

I used only the Natural populations for this: 

46 indivs, 7 pops. 

####1. DAPC

/Users/alexjvr/2016RADAnalysis/Bombina/BV234/Analyses_20161128/DAPC.NatPops

tutorial-dapc: A tutorial for Discriminant Analysis of Principal Components (DAPC) using adegenet 2.0.0

total variance = (variance between groups) + (variance within groups)

or more simply, denoting X the data matrix:

VAR(X) = B(X) +W(X)

Usual approaches such as Principal Component Analysis (PCA) or Principal Coordinates Analysis (PCoA / MDS) focus on V AR(X). That is, they only describe the global diversity, possibly overlooking differences between groups. On the contrary, DAPC optimizes B(X) while minimizing W(X): it seeks synthetic variables, the discriminant functions, which show differences between groups as best as possible while minimizing variation within clusters.


1. Convert input to str format

Using pgdspider

Create a file with all the pop names. Use a *spid file from one of the previous conversions


2. Read into R

```
library(adegenet)

BV46 <- read.structure("BV46.Naturalponds.1665.str")

 How many genotypes are there? 46

 How many markers are there? 1665

 Which column contains labels for genotypes ('0' if absent)? 1

 Which column contains the population factor ('0' if absent)? 2

 Which other optional columns should be read (press 'return' when done)? 1: 

 Which row contains the marker names ('0' if absent)? 1

 Are genotypes coded by a single row (y/n)? n

 Converting data from a STRUCTURE .stru file to a genind object... 
```

3. estimate the number of clusters

Using k-means. Which finds the number of clusters with minimises W(X) and maximises B(X). Compare using BIC

Run algorithm on PCA transformed data. I.e. reduce the dataset so that it can run faster.

```
grp.BV46 <- find.clusters(BV.46, max.n.clust=40)

> choose nr of PCs: 35  ##I try to keep all the PCs

> choose k: 2 ##see figure below


names.7 <- c("DE.B", "DE.K", "DE.W", "Sk.SF", "SK.SL", "Upp.Gra", "Upp.K", "Upp.O", "Um.Gr", "Um.Taf", "LT1", "LT3", "Kir.G", "Kir.L", "FIN")
names.15 <- as.character(names.15)
table.value(table(pop(BV46), grp.BV46$grp), col.lab=paste("inf", 1:6), row.lab=1:10)

dapc1.BV46 <- dapc(BV46, grp.BV46$grp)
scatter(dapc1.BV46)
```

####2. sPCA




####3. TESS3



####4. IBD



####5. AMOVA





###3. Barriers to geneflow


####1. EEMS



####2. Model




###4. Colonisation of Artificial Ponds


####1. PCA


####2. IBD within region


####3. AMOVA within region


####4. Admixture analysis within region


####5. EEMS within region

####6. 
